### PR TITLE
Fix RuntimeInformaiton.IsOSPlatform(OSPlatform.OSX) guard

### PR DESCRIPTION
**Pull request by buyaa-n**
Fixes https://github.com/dotnet/roslyn-analyzers/issues/5205

Because `OSX` is the alias of `MacOS` we are replacing the `OSX` with `MacOS` while parsing the OS platform attributes

https://github.com/dotnet/roslyn-analyzers/blob/c5c6a7df6452524557baaa28c47da1d4828c1687/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzer.cs#L1815-L1816

There is no guard for OSX in `OperatingSystem` type and the guard `OperatingSystem.IsMacOS()` is working just fine with the above change, though we forgot about `RuntimeInformaiton.IsOSPlatform(OSPlatform.OSX)` can be used as a guard for `macOS/OSX`. This PR is fixing that gap.

Not sure if 6.0 is already closed, but it is good to have in 6.0 if possible @jmarolf @jeffhandley

### CODE REVIEW

**Review by mavasani — APPROVED** [review]

**Review by jeffhandley — APPROVED** [review]
LGTM. I defer to @jmarolf for whether or not he's already snapped into RC2. If not, then I support including this in the snap. If we've already snapped into RC2, then we would need to request tactics approval for either RC2 or GA.

**Comment by jmarolf** [comment]
This is the last change for RC2

**Comment by buyaa-n** [comment]
> This is the last change for RC2

Thanks @jmarolf

**Pull request merged**
